pkg/reconciler/broker/broker.go

```diff
@@ -235,6 +237,7 @@ func (r *Reconciler) reconcileDecouplingTopicAndSubscription(ctx context.Context
 	projectID, err := utils.ProjectID(r.projectID, metadataClient.NewDefaultMetadataClient())
 	if err != nil {
 		logger.Error("Failed to find project id", zap.Error(err))
+		b.Status.MarkTopicUnknown("ProjectIdNotFound", "Failed to find project id: %w", err)
```

Comment:
Technically this is not specific to the Topic. But I am fine with this, especially considering this part may be moved to controller creation in the future.